### PR TITLE
mm/iob: add ostest test

### PR DESCRIPTION
**Pull request by hujun260**
## Summary
testing: add iob test

## Impact
iob

## Testing
ostest

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



No, this PR description does not adequately meet the NuttX requirements.  Here's why:

* **Insufficient Summary:** While it mentions adding configurations to `ostest`, it doesn't explain *why* these changes are necessary.  Is it fixing a bug? Adding a new feature? Improving performance?  What part of `ostest` is affected? How do these configuration options interact and change the behavior?
* **Vague Impact:** Simply stating "iob" isn't helpful.  Be specific about how these IOB configurations impact the user, the build process, hardware, documentation, security, and compatibility.  For example, do these changes increase memory usage? Do they change the behavior of any tests? Do they require any new dependencies?
* **Missing Testing Details:**  "ostest" is not sufficient testing information.  Specify the *host* operating system, compiler, and target architecture and board configuration.  Provide *actual* testing logs (or links to them) demonstrating the behavior before and after the change.  Just saying "ostest" doesn't prove anything.

**Example of how to improve this PR description:**

## Summary

This change adds IOB configurations to the `ostest` example to demonstrate and test the IOB throttling mechanism. This is necessary to verify the correct behavior of IOB throttling under various buffer and chain configurations. Specifically, it enables `CONFIG_IOB_THROTTLE` and sets related parameters like `CONFIG_IOB_NBUFFERS`, `CONFIG_IOB_NCHAINS`, and enables the notifier. This allows testing of the IOB allocation behavior when the number of allocated buffers exceeds the throttle limit.

## Impact

* **Is new feature added? Is existing feature changed?** No new feature is added, but existing `ostest` functionality is extended to demonstrate IOB throttling.
* **Impact on user:**  NO. This change is specific to the `ostest` example and doesn't impact regular users of NuttX.
* **Impact on build:** NO.  No changes to the build process are required.
* **Impact on hardware:** NO.  No specific hardware dependencies are introduced.
* **Impact on documentation:** YES.  The `ostest` documentation should be updated to explain the new configurations and their purpose. (A separate documentation PR will be created for this).
* **Impact on security:** NO.
* **Impact on compatibility:** NO.
* **Anything else to consider?**  None.

## Testing

I confirm that changes are verified on local setup and works as intended:
* **Build Host(s):** Linux (Ubuntu 20.04), x86_64, GCC 9.4.0
* **Target(s):**  qemu-x86_64:nsh

**Testing logs before change:**

```
(Previous ostest output showing IOB allocation without throttling - include relevant snippets)
```

**Testing logs after change:**

```
(New ostest output showing IOB allocation with throttling engaged - demonstrating the intended throttling behavior)
```

**Review by acassis — APPROVED** [review]

**Review by cederom — CHANGES_REQUESTED** [review]
Thank you @hujun260 :-)

Could you also please add Kconfig part so it stays in sync with CMake? :-)

**Review by cederom — APPROVED** [review]
Perfect! Thank you @hujun260 :-)

File: testing/iob/iob_main.c [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+
+#include <nuttx/config.h>
+#include <nuttx/sched.h>
+
+#include <setjmp.h>
+#include <stdarg.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include "../../../nuttx/mm/iob/iob.h"
```
Comment (xiaoxiang781216) [inline testing/iob/iob_main.c:33]: add search path instead

File: testing/iob/Kconfig [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#
+# For a description of the syntax of this configuration file,
+# see the file kconfig-language.txt in the NuttX tools repository.
+#
+
+config TESTING_IOB
+	tristate "\"Test iob\" testing"
+	default n
+	select MM_IOB
```
Comment (xiaoxiang781216) [inline testing/iob/Kconfig:9]:
```suggestion
	depends on MM_IOB
```

**Review by acassis — APPROVED** [review]

**Pull request merged**